src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -26,13 +34,79 @@ const ReimbursementRequestTable = ({
   const displayedReimbursementRequests =
     tabValue === 1 && allReimbursementRequests ? allReimbursementRequests : userReimbursementRequests;
 
-  const rows = displayedReimbursementRequests
-    .map(createReimbursementRequestRowData)
-    .sort((a, b) => (a.dateSubmitted > b.dateSubmitted ? -1 : 1));
+  // function getComparator<Key extends keyof any>(
+  //   order: Order,
+  //   orderBy: Key
+  // ): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  //   return order === 'desc' ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+  // }
+
+  function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+    return 0;
+  }
+
+  function statusDescendingComparator(a: ReimbursementStatusType, b: ReimbursementStatusType): number {
+    const statusOrder = new Map<ReimbursementStatusType, number>([
+      [ReimbursementStatusType.PENDING_FINANCE, 1],
+      [ReimbursementStatusType.SABO_SUBMITTED, 2],
+      [ReimbursementStatusType.ADVISOR_APPROVED, 3],
+      [ReimbursementStatusType.DENIED, 5]
+    ]);
+
+    let bConverted = statusOrder.get(b);
+    let aConverted = statusOrder.get(a);
+
+    if (bConverted !== undefined && aConverted !== undefined) {
+      if (bConverted < aConverted) {
+        return -1;
+      }
+      if (bConverted > aConverted) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  function vendorDescendingComparator(a: Vendor, b: Vendor): number {
+    if (b.name < a.name) {
+      return -1;
+    }
+    if (b.name > a.name) {
+      return 1;
+    }
+    return 0;
+  }
+
+  const rows = displayedReimbursementRequests.map(createReimbursementRequestRowData).sort((a, b) => {
+    if (orderBy === 'vendor') {
+      return order === 'desc'
+        ? vendorDescendingComparator(a.vendor, b.vendor)
+        : -vendorDescendingComparator(a.vendor, b.vendor);
+    }
+    if (orderBy === 'status') {
+      return order === 'desc'
+        ? statusDescendingComparator(a.status, b.status)
+        : -statusDescendingComparator(a.status, b.status);
+    }
+    return order === 'desc' ? descendingComparator(a, b, orderBy) : -descendingComparator(a, b, orderBy);
+  });
 
   const tabs = [{ label: 'My Requests', value: 0 }];
   if (canViewAllReimbursementRequests) tabs.push({ label: 'All Club Requests', value: 1 });
 
+  const handleRequestSort = (property: ColumnNames) => (event: React.MouseEvent<unknown>) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+    console.log(order, orderBy);
```

Comment:
Let's remove this console.log statement since we don't want to have it in our codebase.